CraftTweaker2-API/src/main/java/crafttweaker/api/world/IWorld.java

```diff
@@ -104,7 +104,25 @@ default IRayTraceResult rayTraceBlocks(IVector3d begin, IVector3d ray, @Optional
 
     @ZenMethod
     default IItemStack getPickedBlock(IBlockPos pos, IRayTraceResult rayTraceResult, IPlayer player) {
-        CraftTweakerAPI.logError(this.getClass().getName() + " does not override IWorld.getPickedBlock!");
+        CraftTweakerAPI.logError(this.getClass().getName() + " does not override IWorld.getPickedBlock, tell the author to fix that.");
+        return null;
+    }
+
+    @ZenMethod
+    default IExplosion createExplosion(IEntity exploder, double x, double y, double z, float strength, boolean causesFire, boolean damagesTerrain) {

```

Comment:
As you know, the `exploder` parameter in Java method is Nullable, but according to the my previous working, the parameter cannot be null in zenscript. And the `@Optional` annotation only can be used to the last parameter. Maybe you can overload the method , which omits the `exploder` parameter?

Reply:
Would changing IEntity to IAny and then checking if the object is an instanceof IEntity work instead?

Reply:
IAny? I wonder whether it accepts null or not. I mean overloading the method. Add
```java
 @ZenMethod
    default IExplosion createExplosion(IEntity exploder, double x, double y, double z, float strength, boolean causesFire, boolean damagesTerrain) {
    this.createExplosion(null, x, y, z, strength, causesFire, damagesTerrain);
}
```

Reply:
Yeah I know what overloading is, but it's more convenient to just be able to use null at the beginning to explicitly say no entity

Reply:
![image](https://user-images.githubusercontent.com/32407067/97064979-c3796c80-156f-11eb-858d-c5eb66114457.png)

apparently null works with non-primitives